mlir/lib/Dialect/SparseTensor/Transforms/Utils/IterationGraphSorter.h

```diff
@@ -76,14 +89,124 @@ class IterationGraphSorter {
   Value out;
   AffineMap loop2OutLvl;
 
-  // Loop itation types;
+  // Loop iteration types;
   SmallVector<utils::IteratorType> iterTypes;
 
   // Adjacency matrix that represents the iteration graph.
   std::vector<std::vector<bool>> itGraph;
 
   // InDegree used for topo sort.
   std::vector<unsigned> inDegree;
+
+public:
+  enum class SparseAccessType {
+    kCompressedSequential,
+    kSingletonScan,
+    kRandomSparse,
+    kDenseSubtensor
+  };
+
+  struct SparseAccessPattern {
+    SparseAccessType type;
+    double expectedSparsity;
+    unsigned memoryIndirections;
+    bool hasGoodLocality;
+  };
+
+private:
+
+  // Add these fields to your LoopMemoryInfo struct:
+  struct LoopMemoryInfo {
+    unsigned totalTensorAccesses;
+    double avgStrideComplexity;
+    double spatialLocalityScore;
+    double temporalReuseScore;
+    double accessPatternRand;
+
+    // Dense tensor access patterns
```

Comment:
Here and elsewhere, period at end of every comment (except for "inline" comments)